items/item-history.js

```diff
@@ -395,22 +376,29 @@ class ItemHistory {
   /**
    * @private
    */
-  _timestampOrNull (result) {
-    return result === null ? null : time.ZonedDateTime.parse(result.getTimestamp().toString());
+  _dateOrNull (result) {
+    return result === null ? null : time.ZonedDateTime.parse(result.toString());
   }
 
   /**
    * @private
    */
-  _dateOrNull (result) {
-    return result === null ? null : time.ZonedDateTime.parse(result.toString());
+  _decimalOrNull (result) {
+    return result === null ? null : result.toBigDecimal();
   }
 
   /**
    * @private
    */
-  _decimalOrNull (result) {
-    return result === null ? null : result.toBigDecimal();
+  _historicItemOrNull (result) {
+    if (result === null) {
+      return null;
+    }
+    return {
+      state: result.getState().toString(),
+      rawState: result.getState(),
```

Comment:
What I'm a bit unsure about is that maximum*/minimum* used to return a number, which the new code does not return anymore, since previousState can essentially be anything, not only a number. AFAICT, there's two options:

- leave it as is (string as convenience + Java state for special needs, which can also e.g. be forwarded to postUpdate())
- add a decimalState (or something) field, which becomes NaN if the state is not a number

Thoughts?

Reply:
I see the problem, that we cannot generally return `state` as a number.
Given the fact that minimum*/maximum* did return a string for a long time and the switch to number was not so long ago, I‘d consider it okay to leave is as it currently is.

Reply:
Is there a way to check whether `result.getState()` is a `DecimalType` in the JS code? If so, I'd simply fill `decimalState` (or `numberState`?) only in that case.

Reply:
Yes, that’s possible.

You need to import the `DecimalType` and perform an `instanceof` check, but given the fact that `QuantityType` and `PercentType` can contain decimal values as well I wonder if it is not easier to perform a `isNaN` on the string state instead of checking the type.

Reply:
I think `QuantityType` and `PercentType` both are subclasses of `DecimalType`, so the instanceof check should work for all 3?

Reply:
I am not sure how the instanceof check behaves for Java classes, you can test by constructing a new QuantityTyoe after importing it and then performing the instanceof check.

Reply:
I went the easier route and checked for `isNaN`.